include/swift/AST/Types.h

```diff
@@ -3177,6 +3177,12 @@ class AnyFunctionType : public TypeBase {
     /// Whether the parameter is marked '@noDerivative'.
     bool isNoDerivative() const { return Flags.isNoDerivative(); }
 
+    /// Whether the parameter might be a semantic result. This includes
+    /// inout parameters
+    bool isSemanticResult() const {
```

Comment:
"Semantic result" is an AutoDiff term, isn't? Should it be named `isAutoDiffSemanticResult`?

Reply:
Looks reasonable, yes.